src/cluster.c

```diff
@@ -3227,11 +3239,9 @@ void clusterBroadcastMessage(clusterMsgSendBlock *msgblock) {
     di = dictGetSafeIterator(server.cluster->nodes);
     while((de = dictNext(di)) != NULL) {
         clusterNode *node = dictGetVal(de);
-
-        if (!node->link) continue;
```

Comment:
can you explain how this check was not sufficient?

Reply:
This code flow was fine, just a refactor. The crash was happening in a similar flow in `clusterPropagatePublish` below.